docs/mongodb/week-3/lesson.md

```diff
@@ -4,309 +4,142 @@ title: MongoDB - 3
-
-Isolation refers to the visibility of transaction operations to other operations happening in parallel.
-
-Durability in relational database systems refers to the property that every transaction that has successfully committed will survive in the face of failure. This usually refers to writing the contents of the committed transaction in persistent storage (hard disk or SDD).
-
-Let's create a new database `mongo_bank` with a collection `accounts`:
-
-`{"collection": "accounts", "account_id": "1", "account_name": "Alex", "account_balance":100}`
-
-`{"collection": "accounts", "account_id": "2", "account_name": "Mary", "account_balance":50}`
-
-This way, we have 2 users with accounts in our bank.
-
-We can now transfer money between th accounts. We will do this using the Mongo shell:
-
-```js
-session = db.getMongo().startSession({ readPreference: { mode: "primary" } });
-
-accountsCollection = session.getDatabase("mongo_bank").accounts;
-
-session.startTransaction({
-  readConcern: { level: "snapshot" },
-  writeConcern: { w: "majority" },
+  app.listen(3000);
 });
-
-try {
-  accountsCollection.update(
-    { account_id: "1" },
-    { $inc: { account_balance: -70 } }
-  );
-  accountsCollection.update(
-    { account_id: "2" },
-    { $inc: { account_balance: 70 } }
-  );
-} catch (error) {
-  print("error:", error);
-}
-source_balance = accountsCollection.findOne({ account_id: "1" })
-  .account_balance;
-target_balance = accountsCollection.findOne({ account_id: "2" })
-  .account_balance;
-if (source_balance < 0 || target_balance < 0) {
-  session.abortTransaction();
-}
-session.commitTransaction();
-session.endSession();
 ```
 
-To which the output should be similar to:
 
-```js
-WriteCommandError({
-  errorLabels: ["TransientTransactionError"],
-  operationTime: Timestamp(1561207870, 4),
-  ok: 0,
-  errmsg: "Transaction 0 has been aborted.",
-  code: 251,
-  codeName: "NoSuchTransaction",
-  $clusterTime: {
-    clusterTime: Timestamp(1561207870, 4),
-    signature: {
-      hash: BinData(0, "1x7oLuNzWehAWylVA/E9tdXLuFA="),
-      keyId: NumberLong("6655324027294318593"),
-    },
-  },
-});
-```
+## Cycle outline
 
-Database operations need to be inside our session if we want to benefit from transactions
+Thought experiment
+------------------
 
-#### Exercise 3
+Imagine an app (Netflix, Amazon) is showing you a list of things (movies, products) etc.
 
-Transfer 40 monads from account 1 to account 2. Observe how it works perfectly fine.
+Trace the journey of the request:
 
-## MongoDB Project
+BROWSER -> REACT -> HTTP -> SERVER -> MONGODB -> SERVER -> HTTP -> REACT -> BROWSER
 
-Based on the data from the AirBnB listings and reviews collection, we want to create a front end for it.
+Learners can use any method they like to describe each stage, it could be a diagram or code snippets if they prefer.
```

Comment:
A couple of possible presentations of such flows:
[Full-stack CRUD flow - Neill.pdf](https://github.com/CodeYourFuture/syllabus/files/5364632/Full-stack.CRUD.flow.-.Neill.pdf) (available as google sheet at https://docs.google.com/spreadsheets/d/1yxMpMPd0f9T_c1s2h9KRQ_DnYVJ6Vym_23bRS7ZeBRg/edit#gid=0 - but I appreciate we'd want to put it in markdown)


And this one from @DomVinyard 
![image](https://user-images.githubusercontent.com/69844/95738099-9cbe5a80-0c80-11eb-8502-390c42836a96.png)

Reply:
oops i guess these should go in the corresponding instructor notes document.